spec/contracts/user_preferences/params_contract_spec.rb

```diff
@@ -86,6 +86,26 @@
       it_behaves_like 'contract is valid'
     end
 
+    context 'when project setting with start_date, due_date and overdue set' do
```

Comment:
A few test cases seem to be missing:
- Specs for global date alerts.
- Specs when some of date fields are missing from the notification. (ie: start_date missing)
- Specs when a date field is missing and another one has an invalid value